hyperspy/_components/expression.py

```diff
@@ -36,19 +45,28 @@ class docstring.
         expression: str
             Component function in SymPy text expression format. See the SymPy
             documentation for details. The only additional constraint is that
-            the variable must be `x`. Also, in `module` is "numexpr" the
+            the variable(s) must be `x` (for 1D components) or `x` and `y` for
+            2D components. Also, if `module` is "numexpr" the
             functions are limited to those that numexpr support. See its
             documentation for details.
         name : str
             Name of the component.
         position: str, optional
             The parameter name that defines the position of the component if
             applicable. It enables adjusting the position of the component
-            interactively in a model.
+            interactively in a model. 2D components require passing a tuple
+            with the name of the two parameters e.g. `("x0", "y0")`.
         module: {"numpy", "numexpr"}, default "numpy"
```

Comment:
Grammar:

"It enables interactive adjustment of the position of the component in the model. For 2D components, a tuple must be passed with the name of the two parameters e.g. `("x0", "y0")`.